src/main/msc/emfat_file.c

```diff
@@ -0,0 +1,351 @@
+    0x91,0x7f,0xb6,0x9f,0x27,0xbc,0x65,0x29,0x38,0x2c,0x39,0x8d,0x6a,0x56,0xc5,0x13,0xbe,0x11,0x1a,0x08,0x37,0xf7,0xac,0x57,0x57,0x63,0xfa,0x66,0x4d,0x4a,0x09,0xce,
+    0x58,0x1f,0x16,0xa1,0x4c,0x01,0x72,0x11,0x92,0x08,0x0c,0xe8,0x72,0x3b,0x2e,0x90,0x8f,0xc2,0xf3,0xc6,0x64,0xd8,0x4a,0x8c,0x29,0x1f,0x2c,0xfe,0x01,0x01,0x45,0xfe,
+    0x9b,0xe6,0x85,0xd5,0x48,0x46,0x46,0x5f,0x48,0xd2,0xc4,0xd9,0x60,0x49,0x4a,0x36,0x8a,0x38,0x58,0xcf,0x01,0x01,0x01,0xfe,0xbd,0x82,0x23,0xd4,0x3d,0x50,0x44,0x44,
+    0x44,0x46,0x5f,0x3d,0x35,0x64,0xa0,0xfe,0x9f,0x2a,0xfe,0x01,0x01,0x01,0x01,0x45,0xfe,0xf1,0xd6,0xfb,0x8e,0xec,0xf4,0x46,0x44,0x44,0x44,0x48,0x4e,0xee,0x81,0x09,
+    0x68,0xfe,0x4b,0x01,0x01,0x01,0x01,0x01,0x45,0xfe,0x23,0xa8,0x51,0xf2,0x83,0x2e,0x44,0x44,0x44,0x46,0x0a,0x60,0x49,0x4a,0xfe,0x4b,0x01,0x01,0x01,0x01,0x01,0x01,
+    0x01,0x45,0xfe,0x32,0x51,0x51,0xea,0x2e,0x46,0x44,0x44,0x50,0x89,0xda,0x28,0xfe,0x4b,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x45,0x0c,0x7e,0x32,0xc1,0x3b,
+    0xc0,0x50,0x46,0x48,0xbf,0xc7,0xfe,0x04,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x45,0xfe,0xec,0xf0,0x9b,0x17,0x08,0xe7,0x17,0x78,0x01,0x01,
+    0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x45,0x4f,0x62,0x62,0x4f,0x45,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,0xff,0x81,0xff,
+    0x00,0xfe,0x00,0x7f,0x00,0xf8,0x00,0x1f,0x00,0xf0,0x00,0x0f,0x00,0xe0,0x00,0x07,0x00,0xc0,0x00,0x03,0x00,0xc0,0x00,0x03,0x00,0x80,0x00,0x01,0x00,0x80,0x00,0x01,
+    0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x80,0x00,0x01,0x00,0x80,0x00,0x01,
+    0x00,0xc0,0x00,0x03,0x00,0xc0,0x00,0x03,0x00,0xe0,0x00,0x07,0x00,0xf0,0x00,0x0f,0x00,0xf8,0x00,0x1f,0x00,0xfe,0x00,0x7f,0x00,0xff,0x81,0xff,0x00
+#endif
+};
+
+USB_OTG_CORE_HANDLE USB_OTG_dev;
+
+emfat_t emfat;
+
+void autorun_read_proc(uint8_t *dest, int size, uint32_t offset, size_t userdata);
+void icon_read_proc(uint8_t *dest, int size, uint32_t offset, size_t userdata);
+void readme_read_proc(uint8_t *dest, int size, uint32_t offset, size_t userdata);
+void bblog_read_proc(uint8_t *dest, int size, uint32_t offset, size_t userdata);
+
+#define CMA_TIME EMFAT_ENCODE_CMA_TIME(2,4,2017, 13,0,0)
+#define CMA { CMA_TIME, CMA_TIME, CMA_TIME }
+
+static emfat_entry_t entries[] =
+{
+    // name          dir    lvl offset  size             max_size        user  time  read               write
+    { "",            true,  0,  0,      0,               0,              0,    CMA,  NULL,              NULL, { 0 } }, // 0 root
+    { "bf.bbl",      false, 1,  0,      0,               0,              0,    CMA,  bblog_read_proc,   NULL, { 0 } }, // 1 bblog.txt
+    { "autorun.inf", false, 1,  0,      AUTORUN_SIZE,    AUTORUN_SIZE,   0,    CMA,  autorun_read_proc, NULL, { 0 } }, // 2 autorun.inf
+    { "icon.ico",    false, 1,  0,      ICON_SIZE,       ICON_SIZE,      0,    CMA,  icon_read_proc,    NULL, { 0 } }, // 3 icon.ico
```

Comment:
Is there any value in having this icon? Probably the same for `autorun.inf`.

Reply:
Contents of the emulated disk can be tuned when going final production. Current contents are here to validate the soundness of the content generator, and more (or less) importantly ... to have some fun 😄 

<img width="227" alt="win7_64__for_" src="https://user-images.githubusercontent.com/14850998/38463148-f5f40238-3b2e-11e8-8fe5-f2181592bcef.png">